components/validate.js

```diff
@@ -0,0 +1,77 @@
+class Validator {
+    validateSet(set) {
+        var durationCheck = false;
+        var notesCheck = false;
+        var repetitionCheck = false;
+        var weightCheck = false;
+        if (set != undefined && set != null) {
+            if (set.duration > 0) {
+                durationCheck == true;
+            }
+            if (set.notes.length < 100) {
```

Comment:
Since notes field is optional, I would rather make `notesCheck` to be true and check if its length is larger than 100.